core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -81,6 +81,26 @@ public static TableMetadata newTableMetadata(
         schema, spec, sortOrder, location, unreservedProperties(properties), formatVersion);
   }
 
+  public static TableMetadata newTableMetadataWithDefaultProperties(
+      Schema schema,
+      PartitionSpec spec,
+      SortOrder sortOrder,
+      String location,
+      Map<String, String> properties) {
+    Map<String, String> defaults = Maps.newHashMap();
+    defaults.put(
+        TableProperties.PARQUET_COMPRESSION,
+        TableProperties.PARQUET_COMPRESSION_DEFAULT_SINCE_1_4_0);
+    defaults.put(
+        TableProperties.DELETE_PARQUET_COMPRESSION,
+        TableProperties.PARQUET_COMPRESSION_DEFAULT_SINCE_1_4_0);
+    defaults.putAll(unreservedProperties(properties));
+    int formatVersion =
```

Comment:
I wonder whether we should add a method for fetching the format version, it is used in 3 places now.

```
private static int formatVersion(Map<String, String> properties) {
  return PropertyUtil.propertyAsInt(
      properties, TableProperties.FORMAT_VERSION, DEFAULT_TABLE_FORMAT_VERSION);
}
```

Reply:
I think its a good idea, I no longer make a new method though now so it may be unrelated as a change.